openpype/pipeline/publish/publish_plugins.py

```diff
@@ -310,43 +309,21 @@ class ColormanagedPyblishPluginMixin(object):
         ext.lstrip(".") for ext in IMAGE_EXTENSIONS.union(VIDEO_EXTENSIONS)
     )
 
-    @staticmethod
-    def get_colorspace_settings(context):
-        """Returns solved settings for the host context.
+    def get_colorspace_settings(self, context):
+        """[Deprecated] Returns solved settings for the host context.
 
         Args:
             context (publish.Context): publishing context
 
         Returns:
             tuple | bool: config, file rules or None
         """
-        if "imageioSettings" in context.data:
-            return context.data["imageioSettings"]
-
-        project_name = context.data["projectName"]
-        host_name = context.data["hostName"]
-        anatomy_data = context.data["anatomyData"]
-        project_settings_ = context.data["project_settings"]
-
-        config_data = get_imageio_config(
-            project_name, host_name,
-            project_settings=project_settings_,
-            anatomy_data=anatomy_data
-        )
-
-        # in case host color management is not enabled
-        if not config_data:
-            return None
-
-        file_rules = get_imageio_file_rules(
-            project_name, host_name,
-            project_settings=project_settings_
+        self.log.warning(
+            "'get_colorspace_settings' is deprecated, "
+            "use 'get_colorspace_settings_from_publish_context' instead"
         )
-
-        # caching settings for future instance processing
-        context.data["imageioSettings"] = (config_data, file_rules)
-
-        return config_data, file_rules
+        # TODO: for backward compatibility, remove in future
+        return get_colorspace_settings_from_publish_context(context.data)
```

Comment:
I'm not sure about this. That'd basically mean deprecating the full mixin, right? Since just like this function the `set_representation_colorspace` method is also just calling a single function now too.

Reply:
Agree this is needs to be done differently. Look to https://github.com/ynput/OpenPype/pull/5498 the rework.